src/event/log_event.rs

```diff
@@ -290,81 +326,68 @@ impl<K: AsRef<str>, V: Into<Value>> FromIterator<(K, V)> for LogEvent {
     }
 }
 
+/// Converts event into an iterator over top-level key/value pairs.
+impl IntoIterator for LogEvent {
+    type Item = (String, Value);
+    type IntoIter = std::collections::btree_map::IntoIter<String, Value>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        let fields: BTreeMap<_, _> = self.into();
+        fields.into_iter()
+    }
+}
+
```

Comment:
Can this reasonably be avoided? This implicitly drops the metadata, which may have hidden consequences. I'd like to eventually drop the `From<LogEvent>` implementations as well as part of the end-to-end metadata work.

Reply:
Yes. It was only used it one place ([here](https://github.com/timberio/vector/blob/fa69c09fea5c98dfe50938a7bddef7f2350eec32/src/event/log_event.rs#L354-L359)). I've removed the `IntoIterator` and updated that code to iterate through the map directly. I've also changed it so it doesn't wipe out the entire event when you `del(.)` in remap. Now it just clears all the fields and keeps the metadata intact, which I presume is correct?

Reply:
Yes, I think that is the right thing to do.